### PR TITLE
CMake: Fix dll install location

### PR DESCRIPTION
**Pull request by jschueller**
This fixes dll search on conda where dlls are expected to be installed in prefix/bin (pretty standard layout)
static or import libs still go to prefix/lib

### CODE REVIEW

**Comment by mmuetzel** [comment]
The same issue is solved slightly differently with a patch in MSYS2:
https://github.com/msys2/MINGW-packages/blob/8ff52af093c2d67d665e147d79049819e902e2bb/mingw-w64-sundials/0003-sundials-use-default-installdirs.patch

But admittedly, I prefer your proposed change.
Looks good to me. 👍

**Comment by jschueller** [comment]
didnt know that you could omit DESTINATION like that, but that's only since cmake 3.14

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## Changes to SUNDIALS in release X.Y.Z
 
+Fixed the runtime library installation path for windows systems. This fix changes the
```
Comment (balos1) [inline CHANGELOG.md:5]:
```suggestion
Fixed the runtime library installation path for windows systems. This fix changes the 
default library installation path from `CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_LIBDIR` to
`CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_BINDIR`.
```

Comment (balos1) [reply CHANGELOG.md:5]: Markdown formatting should use single backticks for inline code. Please also add this to `docs/shared/RecentChanges.rst` as well (which expects the double backticks).

**Review by balos1 — CHANGES_REQUESTED** [review]
Just a couple request w.r.t. the change notes. Please also sign your commit as per our [Contributing guide](https://github.com/LLNL/sundials/blob/main/CONTRIBUTING.md).

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,10 @@
...
+default library installation path from ``CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_LIBDIR`` to
+``CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_BINDIR``.
```
Comment (balos1) [inline CHANGELOG.md:7]:
```suggestion
`CMAKE_INSTALL_PREFIX/CMAKE_INSTALL_BINDIR`.
```

File: CHANGELOG.md (hunk 1 above)
Comment (jschueller) [reply CHANGELOG.md:5]: ok

**Review by balos1 — APPROVED** [review]

**Pull request merged**